packages/ibm-products/src/components/Datagrid/Datagrid/DatagridSelectAll.tsx

```diff
@@ -79,12 +79,13 @@ const SelectAll = (datagridState: DataGridState) => {
         target: { checked: false },
       } as any);
     }
-    handleSelectAllRowData({
-      dispatch,
-      rows,
-      getRowId,
-      isChecked: event.target.checked,
-      indeterminate,
+    // deselect all rows first
+    (getToggleAllRowsSelectedProps as any)?.()?.onChange({
+      target: { checked: false },
+    });
+    // select all row on current page
+    (getToggleAllPageRowsSelectedProps as any)().onChange({
+      target: { checked: true },
```

Comment:
I think this might be too extensive. A better way is to only select the row if the row isn't disabled. What do you think?

Reply:
Hi @AlexanderMelox ,

I have noticed a few other inconsistencies, with this select-all feature
1) while we have selections on other pages, and if we click on the indeterminate state on any page, it deselects other pages as well, while the expected behavior is, that it should deselect the rows only on the current page. (will be fixed in this pr)
2) `Select all` button in the batch actions, selects the disabled rows, on other pages, maybe [this](https://github.com/carbon-design-system/ibm-products/blob/main/packages/ibm-products/src/components/Datagrid/Datagrid/addons/stateReducer.js#L116) action needs a refactor (will be raising a separate issue for this)

I have converted this pr to a draft, to recreate a multi-use case on the pagination story, to verify the changes, and I'll be reverting the story once its ready.